core/object/object.cpp

```diff
@@ -2026,18 +2037,34 @@ uint32_t Object::get_edited_version() const {
 }
 #endif
 
+const GDType &Object::get_gdtype() const {
+	if (unlikely(!_gdtype_ptr)) {
+		// While class is initializing / deinitializing, constructors and destructors
+		// need access to the proper type at the proper stage.
+		return _get_typev();
+	}
+	return *_gdtype_ptr;
+}
+
+bool Object::is_class(const String &p_class) const {
```

Comment:
Wait, so, what is `is_class` exactly (i.e. what are we trying to optimize here)? The name seems... not great, to say the least, and the header doesn't have a doc.

Is it an `instanceof` operation? i.e. `this->is_class("Node") === this instanceof Node`?

Reply:
In the theme of promoting the type-first architecture, maybe this PR should introduce `Object::is_subtype_of(GDType const&)` (optionally with an `is_subtype_of(String)` overload that resolves the type and calls the other one).

I realize right now we do a lot of "pass the StringName around and do lookups with it", but we ought to be in the mindset of "Pass the GDType around and do stuff with that directly"